src/connector/src/parser/additional_columns.rs

```diff
@@ -87,75 +88,172 @@ fn kafka_compatible_column_vec() -> Vec<(&'static str, CompatibleAdditionalColum
                     column_desc: ColumnDesc::named_with_additional_column(
                         name,
                         id,
-                        DataType::Int64,
+                        DataType::Varchar,
                         AdditionalColumnType::Offset,
                     ),
                     is_hidden: false,
                 }
             }),
         ),
-        // Todo(tabVersion): add header column desc
-        // (
-        //     "header",
-        //     Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
-        //         ColumnCatalog {
-        //             column_desc: ColumnDesc::named(name, id, DataType::List(
-        //
-        //             )),
-        //             is_hidden: false,
-        //         }
-        //     }),
-        // ),
+        (
+            "header", // type: struct<key varchar, value bytea>[]
+            Box::new(|id: ColumnId, name: &str| -> ColumnCatalog {
+                ColumnCatalog {
+                    column_desc: ColumnDesc::named_with_additional_column(
+                        name,
+                        id,
+                        DataType::List(get_kafka_header_item_datatype().into()),
+                        AdditionalColumnType::Header,
```

Comment:
Should this be `DataType::Struct` instead?

Reply:
No, Kafka header is a list of kvs, with schema (varchar, bytes). The list can be empty, ie. no header and multiple kvs,

here is an example

https://github.com/risingwavelabs/risingwave/blob/600300a31babec3f4956f088b4f8788cdd104714/scripts/source/prepare_ci_kafka.sh#L79

kcat will generate 101 messages containing header with two kvs (header1, v1) and (header2, v2).